### PR TITLE
✨ Add `checkbox` component

### PR DESCRIPTION
**Pull request by Viicos**
I can't test the storybook locally as running build/watch crashes my computer

### CODE REVIEW

File: src/components/formio/checkbox.stories.tsx [hunk 1]
```diff
@@ -37,6 +37,13 @@ export const WithToolTip: Story = {
   },
 };
 
+export const WithDescription: Story = {
+  args: {
+    label: 'With descipriotn',
```
Comment (sergei-maertens) [inline src/components/formio/checkbox.stories.tsx:42]:
```suggestion
    label: 'With description',
```

File: src/components/formio/checkbox.tsx [hunk 2]
```diff
@@ -4,21 +4,31 @@ import {Field, FormikHandlers, useFormikContext} from 'formik';
...
 import Component from './component';
+import Description from './description';
 import Tooltip from './tooltip';
 
 export interface CheckboxProps {
   name: string;
   label?: React.ReactNode;
+  required?: boolean;
   tooltip?: string;
+  description?: string;
```
Comment (sergei-maertens) [inline src/components/formio/checkbox.tsx:15]: may also be a `React.ReactNode`, right?

File: src/registry/checkbox/edit-validation.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import {IntlShape} from 'react-intl';
+import {z} from 'zod';
+
+import {buildCommonSchema} from '@/registry/validation';
+
+// undefined (optional) for unspecified, otherwise a finite numeric value. Note that
```
Comment (sergei-maertens) [inline src/registry/checkbox/edit-validation.ts:6]: please update the comments when you copy paste :-)

File: src/registry/checkbox/edit-validation.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+// null would be nicer, but formio's schema does not support null for validate.min,
+// validate.max or defaultValue
+const booleanSchema = z.boolean().optional();
+
+const defaultValueSchema = z.object({defaultValue: booleanSchema});
```
Comment (sergei-maertens) [inline src/registry/checkbox/edit-validation.ts:11]: If you set `EditForm.defaultValues = {...}` appropriately, I think it'll be pinned to either `false` or `true` and will add in any missing `defaultValue` key.

I think we can just stick to `buildCommonSchema` for checkbox component  - the UI control is a checkbox anyway which forces boolean value.

**Review by sergei-maertens — CHANGES_REQUESTED** [review]
There's an ESM build error to it seems, check if you can reproduce with `npm run build`

File: src/components/formio/checkbox.tsx (hunk 2 above)
Comment (Viicos) [reply src/components/formio/checkbox.tsx:15]: It is used this way: `{description && <Description text={description} />}` (and in other components as well)

File: src/registry/checkbox/edit-validation.ts (hunk 4 above)
Comment (Viicos) [reply src/registry/checkbox/edit-validation.ts:11]: So it resutls in `const schema = (intl: IntlShape) => buildCommonSchema(intl);`?

File: src/components/formio/checkbox.tsx (hunk 2 above)
Comment (sergei-maertens) [reply src/components/formio/checkbox.tsx:15]: yeah, with:

```tsx
interface DescriptionProps {
  text: React.ReactNode;
}
```
but I see it's indeed inconsistent in existing components, so let's leave it a narrow type and widen it if needed.

File: src/registry/checkbox/edit-validation.ts (hunk 4 above)
Comment (sergei-maertens) [reply src/registry/checkbox/edit-validation.ts:9]: yep!

At some point we should set up the `useZodErrorMap` hook from the SDK here too, but it's still a bit mixed to set up more specific error messages that need to be translatable. And we need to check how many translations overlap, potentially that should also become its own library.

**Review by sergei-maertens — APPROVED** [review]

**Pull request merged**